src/charm.py

```diff
@@ -69,6 +69,48 @@ def __init__(self, *args):
         self.legacy_db_relation = DbProvides(self, admin=False)
         self.legacy_db_admin_relation = DbProvides(self, admin=True)
 
+    @property
+    def app_peer_data(self) -> Dict:
```

Comment:
You might want to consider a separate peers.py class in the relations directory in future, since it's technically a relation.

Reply:
Yeah, good idea. In this specific case, it should be changed to use Juju secrets in the future, so I think it can be kept here.